pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -1857,4 +1868,65 @@ private static class QueryServers {
       }
     }
   }
+
+  private BrokerResponseNative processShowTablesQuery(SqlNodeAndOptions sqlNodeAndOptions, HttpHeaders httpHeaders,
+      long compilationStartTimeNs, RequestContext requestContext) {
+    Map<String, String> queryOptions = sqlNodeAndOptions.getOptions();
```

Comment:
Currently the access control is only added on the controller side (`PinotQueryResource`). We should also perform access control on broker (in this class)

Reply:
I've made this change.